### PR TITLE
Add staff hours

### LINKED ISSUE #6: Schema for staff hours
Would replace `~staff/staff_hours.yaml`

### PR DESCRIPTION
**Pull request by redplanks**
Fixes #6

I decided while I was at it to rewrite the format to the one vaibhav was working on, since it looks better  IMO. I'll dig up his PR when rewriting the ocflib code, or if he wants to work on it again, that would be very much appreciated.

I wanted to keep canceling staff hours fast and simple, so I was thinking that `ocflib` would read a file of newline-separated usernames from `~staff` and cancel staff hours from those usernames--if the file was malformed or didn't exist, the ocflib code would assume that no staff hours were canceled.

If you think that making a PR to `etc` is not too high a barrier for canceling, or have a better method in mind, pls suggest.

### CODE REVIEW

**Comment by redplanks** [comment]
Just remembered, I'll need to write a custom test to make sure none of the hour ranges overlap.

File: schemas/staff_hours.schema.json [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      "type": "object",
+      "properties": {
+        "Monday": {
+          "type": [
+            "array",
+            "null"
+          ],
+          "items": {
+            "$ref": "#/definitions/staff_hours_entry"
+          }
```
Comment (cg505) [inline schemas/staff_hours.schema.json]: Can we make a type for staff_hours_day or staff_hours_entry_list or something?

Comment (redplanks) [reply schemas/staff_hours.schema.json]: Do you mean having the days as an array instead of hardcoded properties?

**Comment by redplanks** [comment]
I was thinking it might be hard to display overlapping hour ranges clearly, but maybe if we sort by start hour it'll be ok

File: configs/staff_hours.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+staff-hours:
+  Monday:
+  Tuesday:
+  Wednesday:
+  Thursday:
+  Friday:
+  Saturday:
+  Sunday:
+
+staff-positions:
```
Comment (dkess) [inline configs/staff_hours.yaml]: This is fine for now, but we should plan to replace this with data from officers (see #3)

File: schemas/staff_hours.schema.json (hunk 1 above)
Comment (dkess) [reply schemas/staff_hours.schema.json]: This can be made more concise, see https://github.com/ocf/etc/blob/master/schemas/hours.schema.json#L79-L109

**Comment by dkess** [comment]
I think it's OK to keep cancellations in this file, but I suppose a separate file could work too. Just make sure it doesn't fail silently if the file isn't found or can't be read.
